osu.Game/Screens/OsuScreenStack.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Screens;
+using osu.Game.Graphics.Containers;
+
+namespace osu.Game.Screens
+{
+    public class OsuScreenStack : ScreenStack
+    {
+        [Cached]
+        private BackgroundScreenStack backgroundScreenStack;
+
+        private ParallaxContainer parallaxContainer;
+
+        public OsuScreenStack()
+        {
+            initializeStack();
+        }
+
+        public OsuScreenStack(IScreen baseScreen)
+            : base(baseScreen)
+        {
+            initializeStack();
+        }
+
+        private void initializeStack()
+        {
+            InternalChild = parallaxContainer = new ParallaxContainer
+            {
+                RelativeSizeAxes = Axes.Both,
+                Child = backgroundScreenStack = new BackgroundScreenStack { RelativeSizeAxes = Axes.Both },
+            };
+
+            ScreenPushed += setParallax;
```

Comment:
shouldn't `ScreenExited` also be handled? else parallax is going to get stuck int he wrong state